internal/statement.go

```diff
@@ -145,84 +142,64 @@ func processCreateStmt(conv *Conv, n nodes.CreateStmt) {
+	updateSchema(conv, table, constraints, "CREATE TABLE")
 }
 
-func processColumnDef(conv *Conv, n nodes.ColumnDef, pgTable string) (string, ddl.ColumnDef, pgColDef, []constraint, error) {
+func processColumn(conv *Conv, n nodes.ColumnDef, table string) (string, schema.Column, []constraint, error) {
 	mods := getTypeMods(conv, n.TypeName.Typmods)
 	if n.Colname == nil {
-		return "", ddl.ColumnDef{}, pgColDef{}, nil, fmt.Errorf("colname is nil")
-	}
-	pgCol := *n.Colname
-	spCol, err := GetSpannerCol(conv, pgTable, pgCol, false)
-	if err != nil {
-		return "", ddl.ColumnDef{}, pgColDef{}, nil, fmt.Errorf("can't get Spanner col: %w", err)
+		return "", schema.Column{}, nil, fmt.Errorf("colname is nil")
 	}
+	name := *n.Colname
 	tid, err := getTypeID(n.TypeName.Names.Items)
 	if err != nil {
-		return "", ddl.ColumnDef{}, pgColDef{}, nil, fmt.Errorf("can't get type id for %s: %w", spCol, err)
+		return "", schema.Column{}, nil, fmt.Errorf("can't get type id for %s: %w", name, err)
 	}
-	t, issues := getSpannerType(conv, tid, mods)
-
-	// Treatment of arrays: if the PostgreSQL type has a single array parameter
-	// we map it to a Spanner array. The array bounds themselves are ignored
-	// since Spanner doesn't support them (and PostgreSQL ignores them anyway:
-	// https://www.postgresql.org/docs/9.1/arrays.html).
-	// Spanner does not support multi-dimensional arrays -- we map those
-	// to STRING(MAX).
-	a := getArrayBounds(conv, n.TypeName.ArrayBounds)
-	if len(a) > 1 {
-		t = ddl.String{ddl.MaxLength{}}
-		issues = append(issues, multiDimensionalArray)
-	}
-	constraints := analyzeColDefConstraints(conv, n, pgTable, n.Constraints.Items, pgCol)
-	pgSchema := pgColDef{id: tid, mods: mods, array: a, issues: issues}
-	cd := ddl.ColumnDef{
-		Name:    spCol,
-		T:       t,
-		IsArray: len(a) == 1,
-		Comment: mkColComment(pgCol, printType(pgSchema)),
-	}
-	return spCol, cd, pgSchema, constraints, nil
+	ty := schema.Type{
+		Name:        tid,
+		Mods:        mods,
+		ArrayBounds: getArrayBounds(conv, n.TypeName.ArrayBounds)}
+	return name, schema.Column{Name: name, Type: ty}, analyzeColDefConstraints(conv, n, table, n.Constraints.Items, name), nil
 }
 
 func processInsertStmt(conv *Conv, n nodes.InsertStmt) *copyOrInsert {
 	if n.Relation == nil {
 		logStmtError(conv, n, fmt.Errorf("relation is nil"))
 		return nil
 	}
-	pgTable, spTable, err := getTableName(conv, *n.Relation)
+	table, err := getTableName(conv, *n.Relation)
 	if err != nil {
 		logStmtError(conv, n, fmt.Errorf("can't get table name: %w", err))
 		return nil
 	}
-	conv.statsAddRow(spTable, conv.schemaMode())
-	cols, err := getCols(conv, pgTable, n.Cols.Items)
+	conv.statsAddRow(table, conv.schemaMode())
+	colNames, err := getCols(conv, table, n.Cols.Items)
 	if err != nil {
 		logStmtError(conv, n, fmt.Errorf("can't get col name: %w", err))
-		conv.statsAddBadRow(spTable, conv.schemaMode())
+		conv.statsAddBadRow(table, conv.schemaMode())
```

Comment:
Same issue. Here, the table name is the raw name from the source schema.

Reply:
statement.go is correct (it uses source table; in fact it longer has spTable).
The error was in process.go.